### PR TITLE
sensors: adi_ltc2990: emul: tests: add adltc2990 emul and tests

### PR DESCRIPTION
**Pull request by jilaypandya**
This PR does the following:

1. Introduce emulator and tests for adi,ltc2990 sensor
2. Minor bug fixes discovered from testing

**TEST_OUTPUT :**
```
------ TESTSUITE SUMMARY START ------

SUITE PASS - 100.00% [adltc2990_1_3]: pass = 4, fail = 0, skip = 0, total = 4 duration = 0.000 seconds
 - PASS - [adltc2990_1_3.test_V1_MINUS_V2_VCC] duration = 0.000 seconds
 - PASS - [adltc2990_1_3.test_ambient_temperature] duration = 0.000 seconds
 - PASS - [adltc2990_1_3.test_current] duration = 0.000 seconds
 - PASS - [adltc2990_1_3.test_die_temperature] duration = 0.000 seconds

SUITE PASS - 100.00% [adltc2990_5_3]: pass = 2, fail = 0, skip = 0, total = 2 duration = 0.000 seconds
 - PASS - [adltc2990_5_3.test_ambient_temperature] duration = 0.000 seconds
 - PASS - [adltc2990_5_3.test_die_temperature] duration = 0.000 seconds

SUITE PASS - 100.00% [adltc2990_6_3]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.000 seconds
 - PASS - [adltc2990_6_3.test_current] duration = 0.000 seconds

SUITE PASS - 100.00% [adltc2990_7_3]: pass = 3, fail = 0, skip = 0, total = 3 duration = 0.000 seconds
 - PASS - [adltc2990_7_3.test_V1_V2_V3_V4_VCC] duration = 0.000 seconds
 - PASS - [adltc2990_7_3.test_available_channels] duration = 0.000 seconds
 - PASS - [adltc2990_7_3.test_die_temperature] duration = 0.000 seconds

------ TESTSUITE SUMMARY END ------

===================================================================
PROJECT EXECUTION SUCCESSFUL

Process finished with exit code 0
```

### CODE REVIEW

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please split this commit:
```
sensors: tests: adi: add ltc2990 tests

This commit adds tests for ltc2990 with varying configurations
This commit fixes some minor bugs discovered during testing

Signed-off-by: Jilay Pandya <jilay.pandya@zeiss.com>
```

Into separate commits:
- Bug fix
- Whitespace changes
- New test

**Comment by jilaypandya** [comment]
> Please split this commit:
> 
> ```
> sensors: tests: adi: add ltc2990 tests
> 
> This commit adds tests for ltc2990 with varying configurations
> This commit fixes some minor bugs discovered during testing
> 
> Signed-off-by: Jilay Pandya <jilay.pandya@zeiss.com>
> ```
> 
> Into separate commits:
> 
> * Bug fix
> * Whitespace changes
> * New test

Hi @MaureenHelm, Thanks for the quick review. I have split the PRs :-)

File: drivers/sensor/adltc2990/adltc2990_emul.c [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		return -EIO;
+	}
+	if (FIELD_GET(I2C_MSG_READ, msgs->flags)) {
+		printk("Unexpected read");
+		return -EIO;
+	}
+	if (msgs->len < 1) {
+		printk("Unexpected msg0 length %s", msgs->len);
+		return -EIO;
+	}
```
Comment (MaureenHelm) [inline drivers/sensor/adltc2990/adltc2990_emul.c:91]: Replace `printk` with `LOG_ERR`

File: drivers/sensor/adltc2990/adltc2990.c [hunk 2]
```diff
@@ -253,15 +254,15 @@ static int adltc2990_sample_fetch(const struct device *dev, enum sensor_channel
...
+				 cfg->pins_v1_v2.pins_current_resistor);
 		}
 		if (mode_v3_v4 == VOLTAGE_DIFFERENTIAL) {
 			data->pins_v3_v4_values[0] =
-				(adltc2990_fetch_property_value(dev, VOLTAGE_DIFFERENTIAL, V3) *
-				 ADLTC2990_MICROOHM_CONVERSION_FACTOR) /
-				cfg->pins_v3_v4.pins_current_resistor;
+				adltc2990_fetch_property_value(dev, VOLTAGE_DIFFERENTIAL, V3) *
+				(ADLTC2990_MICROOHM_CONVERSION_FACTOR /
+				 cfg->pins_v3_v4.pins_current_resistor);
```
Comment (MaureenHelm) [inline drivers/sensor/adltc2990/adltc2990.c]: Looks like this change should move from the bug fix commit to the refactor commit

File: drivers/sensor/adltc2990/adltc2990.h [hunk 3]
```diff
@@ -18,14 +19,14 @@ enum adltc2990_monitor_pins {
...
-} adltc2990_monitor_pins;
+};
 
 enum adltc2990_monitoring_type {
 	NOTHING,
 	VOLTAGE_DIFFERENTIAL,
 	VOLTAGE_SINGLEENDED,
 	TEMPERATURE
-} adltc2990_monitoring_type;
+};
```
Comment (MaureenHelm) [inline drivers/sensor/adltc2990/adltc2990.h:29]: Looks like this change should move from the bug fix commit to the refactor commit

File: tests/drivers/sensor/adltc2990/boards/native_posix.overlay [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        temperature-format = <1>;
+        acquistion-format = <1>;
+        measurement-mode = <7 3>;
+        pins-v1-v2-current-resistor = <0>;
+        pins-v3-v4-current-resistor = <0>;
+        pin-v1-voltage-divider-resistors = <500 1000>;
+        pin-v2-voltage-divider-resistors = <110000 100000>;
+        pin-v3-voltage-divider-resistors = <7000 1000>;
+        pin-v4-voltage-divider-resistors = <500 1000>;
+    };
```
Comment (MaureenHelm) [inline tests/drivers/sensor/adltc2990/boards/native_posix.overlay:61]: Not sure why checkpatch isn't catching it, but this file should be indented with tabs, not spaces

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/adltc2990/adltc2990.h (hunk 3 above)
Comment (jilaypandya) [reply drivers/sensor/adltc2990/adltc2990.h:29]: done

File: drivers/sensor/adltc2990/adltc2990_emul.c (hunk 1 above)
Comment (jilaypandya) [reply drivers/sensor/adltc2990/adltc2990_emul.c]: done

File: tests/drivers/sensor/adltc2990/boards/native_posix.overlay (hunk 4 above)
Comment (jilaypandya) [reply tests/drivers/sensor/adltc2990/boards/native_posix.overlay]: corrected

**Review by MaureenHelm — APPROVED** [review]

**Review by aasinclair — APPROVED** [review]

**Comment by jilaypandya** [comment]
Test

**Pull request merged**